Nexpo.Tests/Controllers/CompaniesControllerTest.cs

```diff
@@ -0,0 +1,232 @@
+            var client = application.CreateClient();
+            var token = await Login("admin", client);
+            var dto = new UpdateCompanyDto();
+            dto.Description = "New description";
+            
+            var json = new JsonObject();
+            json.Add("description", dto.Description);
+            var payload = new StringContent(json.ToString(), Encoding.UTF8, "application/json");
+            var response = await client.PutAsync("/api/companies/-1", payload);
+
+            Assert.True(response.StatusCode.Equals(HttpStatusCode.OK), response.StatusCode.ToString());
+
+            string responseText = await response.Content.ReadAsStringAsync();
+            var responseObject = JsonConvert.DeserializeObject<PublicCompanyDto>(responseText);
+         
+            Assert.True(responseObject.Description == "New description", $"Description was actually ({responseObject.Description})");
+            json.Remove("description");
+            json.Add("description", "A fruit company");
+            await client.PutAsync("/api/companies/-1", new StringContent(json.ToString(), Encoding.UTF8, "application/json"));
+        }
+
+        [Fact]
+        public async Task PutCompanyForbidden()
+        {
+            var application = new WebApplicationFactory<Nexpo.Program>();
+            var client = application.CreateClient();
+            var token = await Login("company", client);
+            var dto = new UpdateCompanyDto();
+            dto.Description = "None";
+            
+            var json = new JsonObject();
+            json.Add("description", dto.Description);
+            var payload = new StringContent(json.ToString(), Encoding.UTF8, "application/json");
+            var response = await client.PutAsync("/api/companies/-1", payload);
+
+            Assert.True(response.StatusCode.Equals(HttpStatusCode.Forbidden), response.StatusCode.ToString());
+
+            string responseText = await response.Content.ReadAsStringAsync();
+            var responseObject = JsonConvert.DeserializeObject<PublicCompanyDto>(responseText);
+         
+            Assert.True(responseObject == null, "Object was not null");
+        }
+
+        [Fact]
+        public async Task PutNonExistingCompany()
+        {
+            var application = new WebApplicationFactory<Nexpo.Program>();
+            var client = application.CreateClient();
+            var token = await Login("admin", client);
+            var dto = new UpdateCompanyDto();
+            dto.Description = "None";
+            
+            var json = new JsonObject();
+            json.Add("description", dto.Description);
+            var payload = new StringContent(json.ToString(), Encoding.UTF8, "application/json");
+            var response = await client.PutAsync("/api/companies/-5", payload);
+
+            Assert.True(response.StatusCode.Equals(HttpStatusCode.InternalServerError), response.StatusCode.ToString());
```

Comment:
HttpStatusCode Internal Server Error indicates that something isn't working as intended on the server. The correct response should be "NotFound".

Reply:
This issue has been fixed with the latest commit. :)